### PR TITLE
Issue/6482

### PR DESCRIPTION
**Pull request by paulp**
Review by @odersky if possible, or @adriaanm or @retronym.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala [hunk 1]
```diff
@@ -111,20 +111,51 @@ abstract class ExtensionMethods extends Transform with TypingTransformers {
...
+    *  some higher level facilities.
+    */
     def extensionMethInfo(extensionMeth: Symbol, origInfo: Type, clazz: Symbol): Type = {
-      // No variance for method type parameters
-      var newTypeParams = cloneSymbolsAtOwner(clazz.typeParams, extensionMeth) map (_ resetFlag COVARIANT | CONTRAVARIANT)
-      val thisParamType = appliedType(clazz.typeConstructor, newTypeParams map (_.tpeHK))
+      val GenPolyType(tparamsFromMethod, methodResult) = origInfo cloneInfo extensionMeth
+      // Start with the class type parameters - clones will be method type parameters
+      // so must drop their variance.
+      var tparamsFromClass = cloneSymbolsAtOwner(clazz.typeParams, extensionMeth) map (_ resetFlag COVARIANT | CONTRAVARIANT)
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala]: why var?

Comment (paulp) [inline src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala]: Because I inherited it from the previous implementation and didn't notice.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by adriaanm** [comment]
LGTM

**Pull request merged**